### PR TITLE
Add resource for Application Insights and store to vault

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Router: Create a new Micro service](https://tools.hmcts.net/jira/browse/BPS-794)

### Change description ###

Name reference is ~picked up from Jenkins pipeline. It errors to fetch it but is soft dependency so not a failure to deploy.~ standard one

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -32,3 +32,18 @@ data "azurerm_key_vault" "key_vault" {
...
+resource "azurerm_application_insights" "appinsights" {
+  name                = "${var.product}-${var.env}"
+  location            = "${var.location}"
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+  application_type    = "${var.application_type}"
+  tags                = "${var.common_tags}"
+}
+
+# store app insights key in key vault
+resource "azurerm_key_vault_secret" "bulk_scan_orchestrator_app_s2s_secret" {
```
Comment (nitinprabhu) [inline infrastructure/main.tf]: name should be app insights and not secret ?

File: infrastructure/main.tf [hunk 2]
```diff
@@ -32,3 +32,18 @@ data "azurerm_key_vault" "key_vault" {
...
+  name         = "AppInsightsInstrumentationKey"
```
Comment (nitinprabhu) [inline infrastructure/main.tf]: can we plz name it `app-insights-instrumentation-key` ?

File: infrastructure/main.tf (hunk 1 above)
Comment (doncem) [reply infrastructure/main.tf]: 🙊 . let's agree together - you didn't see it :D

File: infrastructure/main.tf (hunk 2 above)
Comment (doncem) [reply infrastructure/main.tf]: and how about jenkins pipeline using this? at first i did put in what you suggest

Comment (nitinprabhu) [reply infrastructure/main.tf]: that would be set in bootstrap.yaml ?

**Pull request merged**